fastdom.js

```diff
@@ -120,28 +124,51 @@ function scheduleFlush(fastdom) {
 }
 
 /**
- * Runs queued `read` and `write` tasks.
+ * Schedules a new read
+ * batch if one isn't pending.
+ *
+ * @private
+ */
+
+function scheduleReadFlush(fastdom) {
+  if (fastdom.scheduledRead) return;
+  fastdom.scheduledRead = true;
+  if(fastdom._isWriting){
+    setTimeout(fastdom._flushReads, 0);
+  } else {
+    fastdom.immediate.then(fastdom._flushReads);
+    //in case we use setTimeout or a bad Promise polyfill, setTimeout can be called later than a rAF.
+    //by additionally using the ordinary scheduleFlush function we make sure that reads are called as early as possible.
+    scheduleFlush(fastdom);
```

Comment:
So effectively: when `.read()` is called 'normally' (ie. not nested inside `.write()`) we have a race between `setTimeout` and `rAF` to flush the queue.

We're assuming that most of the time `setTimeout` will win this race, but I guess this depends how full the task queue is.

Say for example the browser has several expensive graphics tasks scheduled on the main-thread queue, perhaps `setTimeout` might be called after `rAF`? I really don't know!

If `rAF` runs first, are we effectively running the same logic as currently on `master`?